common/src/main/kotlin/no/nav/su/se/bakover/common/periode/Periode.kt

```diff
@@ -40,4 +37,30 @@ data class Periode(
         val minusEnDag = Period.ofDays(-1)
         return sluttStart == plussEnDag || sluttStart == minusEnDag || startSlutt == plussEnDag || startSlutt == minusEnDag
     }
+
+    companion object {
+        fun create(fraOgMed: LocalDate, tilOgMed: LocalDate): Periode {
+            return tryCreate(fraOgMed, tilOgMed).getOrHandle { throw IllegalArgumentException(it.toString()) }
+        }
+
+        fun tryCreate(fraOgMed: LocalDate, tilOgMed: LocalDate): Either<UgyldigPeriode, Periode> {
+            if (fraOgMed.year < 2020) { return UgyldigPeriode.FraOgMedDatoKanIkkeVæreFør2020.left() }
```

Comment:
Det kommer en TODO her. Det var ikke så enkelt å endre disse reglene for Periode, siden den brukes i forbindelse med Grunnbeløp-testene. Og de bruker tidligere år for å verfisere at vi beregner riktig dersom grunnbeløpet endrer seg ila. en stønadsperiode. Muligens gir det mer mening å lage en type som heter Stønadsperiode som inneholder år og månedsvalideringene. For så å mappe den til en periode, som ikke er like streng :shrug:

Reply:
Vi har nå ryddet opp ganske mye rundt dette. Krysser fingrene for at det blir lettere å teste / endre nå.